examples/cpp/modes/goto_global/include/mode.hpp

```diff
@@ -0,0 +1,224 @@
+/****************************************************************************
+ * Copyright (c) 2024 PX4 Development Team.
+ * SPDX-License-Identifier: BSD-3-Clause
+ ****************************************************************************/
+#pragma once
+
+#include <px4_ros2/components/mode.hpp>
+#include <px4_ros2/control/setpoint_types/goto.hpp>
+#include <px4_ros2/odometry/global_position.hpp>
+#include <px4_ros2/utils/geometry.hpp>
+#include <px4_ros2/utils/geodesic.hpp>
+#include <px4_msgs/msg/vehicle_attitude.hpp>
+
+#include <rclcpp/rclcpp.hpp>
+#include <Eigen/Core>
+#include <algorithm>
+
+static const std::string kName = "Go-to Global Example";
+static const std::string kNodeName = "example_mode_goto_global";
+
+using namespace px4_ros2::literals; // NOLINT
+
+class FlightModeTest : public px4_ros2::ModeBase
+{
+public:
+  explicit FlightModeTest(rclcpp::Node & node)
+  : ModeBase(node, kName)
+  {
+    _goto_setpoint = std::make_shared<px4_ros2::GotoGlobalSetpointType>(*this);
+
+    _vehicle_global_position = std::make_shared<px4_ros2::OdometryGlobalPosition>(*this);
+
+    _vehicle_attitude_sub = node.create_subscription<px4_msgs::msg::VehicleAttitude>(
+      "/fmu/out/vehicle_attitude", rclcpp::QoS(1).best_effort(),
+      [this](px4_msgs::msg::VehicleAttitude::UniquePtr msg) {
+        updateVehicleHeading(msg);
+      }
+    );
```

Comment:
We should have some odometry class for attitude (we can add that later on)

Reply:
Great point, I'll make a PR for that after this one